src/services/ServerInfo.js

```diff
@@ -44,14 +47,22 @@ module.exports = class ServerInfo {
    * Returns the first IPv4 address of the first connected wifi interface.
    */
   getFirstWifiInterface() {
-    const wifi = require('node-wifi');
-    wifi.init();
-    return wifi.getCurrentConnections().then(
-      (connections) => {
-        let firstWifiConnection = connections[0];
-        let firstWifiInterface = os.networkInterfaces()[firstWifiConnection.iface];
-        return firstWifiInterface.filter((address) => address.family === 'IPv4')[0];
-      });
+    return new Promise((resolve, reject) => {
```

Comment:
I wouldn't wrap it in another promise. Deferring to other promises introduces extra complexity and might result into unexpected issues. For example, in the code below, if `getCurrentConnections()` rejects, its error won't be delegated correctly (since returning a promise in a promise executor alone does not extend the original promise chain, unless `resolve(otherPromise)` is called).

What about returning `Promise.reject(...)` in the `catch` block instead?
```
      try {
        const wifi = require('node-wifi');
        wifi.init();
        return wifi.getCurrentConnections().then((connections) => {
          if(!connections.length) {
            throw new Error('No WiFi interfaces');
          }
          let firstWifiConnection = connections[0];
          let firstWifiInterface = os.networkInterfaces()[firstWifiConnection.iface];
          return firstWifiInterface.filter((address) => address.family === 'IPv4')[0].address;
        });
      } catch(e) {
        return Promise.reject(e.message);
      }
```